db/collection.go

```diff
@@ -1311,7 +1306,7 @@ func (c *collection) tryGetFieldKey(key core.PrimaryDataStoreKey, fieldName stri
 func (c *collection) tryGetSchemaFieldID(fieldName string) (uint32, bool) {
 	for _, field := range c.Schema().Fields {
 		if field.Name == fieldName {
-			if field.IsObject() || field.IsObjectArray() {
```

Comment:
question: Why has this check been removed?  It looks like it is still needed to me.

Reply:
Doing it this way allows using the alias `author` instead of `author_id` to set the relationship.

Reply:
Why do we want `author_id` to set the relationship?

Reply:
That is what we always did. This change allows `author` to be used.

Reply:
Sorry, I got mixed up.

Why do we want `author` to set the relationship?

Reply:
Because `author` is an accepted alias to `author_id`.

Reply:
I'm still not understanding why this change has been made - is there a test that this affects?

todo: The comment on the line below is now out of date, as some `navigational properties` are now matched.

Reply:
It might be a relic from before I fixed everything. It was a change that helped with some tests but I'll revert and see what happens. `author` should still be usable by users regardless.

Reply:
I can confirm my previous comment. It came from a previous change that is no longer needed.

Reply:
reverted ✅